test/rhsmlib/dbus/test_config.py

```diff
@@ -10,64 +10,63 @@
 # Red Hat trademarks are not licensed under GPLv2. No permission is
 # granted to use or replicate Red Hat trademarks that are incorporated
 # in this software or its documentation.
+import tempfile
+from typing import Any, Dict
 
 import dbus
 
 from rhsm.config import RhsmConfigParser
-from rhsmlib.dbus import constants
 from rhsmlib.dbus.objects.config import ConfigDBusObject
-from test.rhsmlib.base import DBusObjectTest, TestUtilsMixin
+from test.rhsmlib.base import DBusServerStubProvider
 
 from test.rhsmlib.services.test_config import TEST_CONFIG
-from test import subman_marker_dbus
 
 
-@subman_marker_dbus
-class TestConfigDBusObject(DBusObjectTest, TestUtilsMixin):
-    def setUp(self):
-        super(TestConfigDBusObject, self).setUp()
-        self.proxy = self.proxy_for(ConfigDBusObject.default_dbus_path)
-        self.interface = dbus.Interface(self.proxy, constants.CONFIG_INTERFACE)
+class TestConfigDBusObject(DBusServerStubProvider):
+    dbus_class = ConfigDBusObject
+    dbus_class_kwargs: Dict[str, Any] = {"parser": None}
 
-    def dbus_objects(self):
-        self.fid = self.write_temp_file(TEST_CONFIG)
-        self.addCleanup(self.fid.close)
-        self.parser = RhsmConfigParser(self.fid.name)
-        return [(ConfigDBusObject, {"parser": self.parser})]
+    config_file = None
+    """Attribute referencing file containing test configuration text."""
 
-    def test_get_all(self):
-        def assertions(*args):
-            result = args[0]
-            self.assertIn("server", result)
+    parser = None
+    """Attribute referencing configuration's parser object."""
 
-        dbus_method_args = [""]
-        self.dbus_request(assertions, self.interface.GetAll, dbus_method_args)
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.config_file = tempfile.NamedTemporaryFile()
+        with open(cls.config_file.name, "w") as handle:
+            handle.write(TEST_CONFIG)
```

Comment:
strictly speaking, `config_file` is already open :stuck_out_tongue: simply write to it, close the fd, and keep the file around